### ISSUE TITLE
Ne faire de requette de création de certificat uniquement si le domaine est fonctionnel

### ISSUE THREAD

**Issue by soul9** [AlternC/alternc-certbot#12]
J'ai du faire ca car on a plein de comptes zombies sur notre hébergement, et le script a l'air de vouloir créer des certificats pour tout domaine, meme les domaines TXT etc.
Donc j'ai diverti /usr/bin/certbot avec dpkg vers /usr/bin/certbot.orig et créé un script /usr/bin/certbot avec grosso modo ca:
```bash
#!/bin/bash
domain=$(echo "$@" |sed -r "s,.*-d ([^ ]+).*,\1,g")

TMPDIR=/var/lib/letsencrypt/.well-known/acme-challenge/
mkdir -p $TMPDIR
tmp=$(mktemp --tmpdir=$TMPDIR)

echo $RANDOM > $tmp
chmod o+r $tmp
# Validate domain, if it is non functional, exit 
if [ "$(wget --timeout=1 --tries=1 -O - -q http://$domain/.well-known/acme-challenge/$(basename $tmp) |tr -d '\n')" != "$(cat $tmp |tr -d '\n')" ]; then
  rm -fr $TMPDIR
  exit 1
fi

rm -fr $TMPDIR

certbot.orig "$@"
```

**Comment by camlafit** [comment]
Hello

I'll check alternc abi, but I'm not sure we can filter domains list.
In fact we can exclude some subdomain type as TXT.

Wget is a workaround which we can't use by default. On self hosting at
least this test could'nt run correctly.

To investigate to evolve :)

**Comment by soul9** [comment]
Idon't understand why you say it would not work correctly on self hosting. Could you please explain?

**Comment by camlafit** [comment]
Hello

Idon't understand why you say it would not work correctly on self hosting.
> Could you please explain?
>
It's about loopback , in some case mainly in self-hosting behind public box
we can't access to domain/public ip (and it's not natable/manageable)

**Comment by soul9** [comment]
Ah, I see. Is there no way to detect whether we are behind NAT or not? There should be an internal ip set, where as on non-nat setup there isn't, no?

**Comment by camlafit** [comment]
Hello

should be solved with f1d8af1070cba6e3f8bfc0974ea6be31f97a70ca
